src/herder/TransactionQueue.cpp

```diff
@@ -930,4 +920,47 @@ TransactionQueue::isFiltered(TransactionFrameBasePtr tx) const
         abort();
     }
 }
+
+TransactionQueue::Limiter::Limiter(int multiplier, LedgerManager& lm)
+    : mPoolLedgerMultiplier(multiplier), mLedgerManager(lm)
+{
+}
+
+size_t
+TransactionQueue::Limiter::maxQueueSizeOps() const
+{
+    size_t maxOpsLedger = mLedgerManager.getLastMaxTxSetSizeOps();
+    maxOpsLedger *= mPoolLedgerMultiplier;
+    return maxOpsLedger;
+}
+
+void
+TransactionQueue::Limiter::addedTransaction(TransactionFrameBasePtr& tx)
+{
+    mQueueSizeOps += tx->getNumOperations();
+}
+
+void
+TransactionQueue::Limiter::removedTransaction(TransactionFrameBasePtr& tx)
+{
+    mQueueSizeOps -= tx->getNumOperations();
+}
+
+bool
+TransactionQueue::Limiter::canAddTx(TransactionFrameBasePtr& tx,
+                                    TransactionFrameBasePtr& oldTx)
+{
+    auto netOps = tx->getNumOperations();
```

Comment:
added an invariant check: it would be a bug to replace a transaction with less operations in the current design

Reply:
I don't think this is right. Fee-bump replace-by-fee can cause a smaller transaction to replace a bigger transaction (the inner transactions do not need to have the same hash). So unless I'm missing something, this breaks fee-bump replace-by-fee.

This revealed to me that we don't test that case right now (eg. use fee-bump to replace 10 operations with 1 operation) so please add a test case for that since it is relevant to these changes.

Reply:
Never mind, you reworked this in a later commit but I responded before I reached that change. I think the new approach is fine. But it is still a good idea to test replacing a bigger transaction with a smaller one, because that is in fact possible.

Reply:
my comment in the PR was stale as well: I realized later that we the only assumption in the limiter is that "old" is already in the queue (it's a replace by X situation). The comment and invariant in the code should be the right one now.

Reply:
Adding additional tests is tracked in https://github.com/stellar/stellar-core/issues/2913